block-data/src/lib.rs

```diff
@@ -38,6 +38,16 @@ pub struct BlockData {
     _type: BlockDataType,
 }
 
+impl BlockDataRead for BlockData {
```

Comment:
Do you know of a good pattern to do this so you don't have to reimpl a trait for every combination of ref/non-ref? Especially annoying when you have a generic in the trait and you end up potentially needing:
```
impl Foo<Bar> for Baz {}
impl Foo<&Bar> for Baz {}
impl Foo<Bar> for &Baz {}
impl Foo<&Bar> for &Baz {}
```
No need to change here since it's only one extra copy and BlockData is not long for this world, but something I've run into in a few places.

Reply:
Unfortunately I don't know a great way to do that, I get why they are separate  since having a trait available on both isn't always ideal anyway but it would be nice to have some way to indicate that you just want it for both.